tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -184,3 +186,33 @@ def test_pantsd_stacktrace_dump(self):
         self.assert_success(self.run_pants_with_workdir(['kill-pantsd'], workdir, pantsd_config))
 
       checker.assert_stopped()
+
+  def test_pantsd_runner_dies_after_failed_run(self):
+    with self.pantsd_test_context() as (workdir, pantsd_config, checker):
+      self.assert_success(self.run_pants_with_workdir(['kill-pantsd'], workdir, pantsd_config))
+      try:
+        # Start pantsd implicitly via a throwaway invocation.
+        self.assert_success(self.run_pants_with_workdir(['help'], workdir, pantsd_config))
+        checker.await_pantsd()
+
+        # Run target that throws an exception in pants.
+        self.assert_failure(
+          self.run_pants_with_workdir(
+            ['bundle', 'testprojects/src/java/org/pantsbuild/testproject/bundle:missing-files'],
+            workdir,
+            pantsd_config)
+        )
+
+        # Check for no stray pantsd-runner prcesses.
+        for proc in psutil.process_iter():
+          try:
+            pinfo = proc.cmdline()
+            self.assertFalse('pantsd-runner' in str(pinfo))
+          except (psutil.NoSuchProcess, psutil.AccessDenied):
+            pass
+
+      finally:
```

Comment:
should this test also attempt a subsequent `self.assert_success()`'d and checked `help` run after the expected-to-fail run to ensure the daemon remains in a good state functionally?